src/mrinufft/trajectories/tools.py

```diff
@@ -700,3 +701,178 @@ def radialize_center(trajectory, nb_samples, in_out=False):
+    center_prop: float or int
+        Proportion of center of kspace to continuouly sample
+    accel: float
+        Undersampling/Acceleration factor
+    pdf: str, optional
+        Probability density function for the remaining samples.
+        "gaussian" (default) or "uniform".
+    rng: random state
+
+    Returns
+    -------
+    np.ndarray: array of size dim_size/accel.
+    """
+    order = VDSorder(order)
+    pdf = VDSpdf(pdf)
+    if accel == 0:
+        return np.arange(dim_size)  # type: ignore
+
+    indexes = list(range(dim_size))
+
+    if not isinstance(center_prop, int):
+        center_prop = int(center_prop * dim_size)
+
+    center_start = (dim_size - center_prop) // 2
+    center_stop = (dim_size + center_prop) // 2
+    center_indexes = indexes[center_start:center_stop]
+    borders = np.asarray([*indexes[:center_start], *indexes[center_stop:]])
+
+    n_samples_borders = (dim_size - len(center_indexes)) // accel
+    if n_samples_borders < 1:
+        raise ValueError(
+            "acceleration factor, center_prop and dimension not compatible."
+            "Edges will not be sampled. "
+        )
+    rng = np.random.default_rng(rng)  # validate get a RNG from a seed or existing rng.
+
+    def _get_samples(p) -> list[int]:
+        p /= np.sum(p)
+        return list(rng.choice(borders, size=n_samples_borders, replace=False, p=p))
+
+    if pdf == VDSpdf.GAUSSIAN:
+        p = norm.pdf(np.linspace(norm.ppf(0.001), norm.ppf(0.999), len(borders)))
+        sampled_in_border = _get_samples(p)
+    elif pdf == VDSpdf.UNIFORM:
+        p = np.ones(len(borders))
+        sampled_in_border = _get_samples(p)
+    elif pdf == VDSpdf.EQUISPACED:
+        sampled_in_border = list(borders[::accel])
+
+    elif isinstance(pdf, np.ndarray):
+        if len(pdf) == dim_size:
+            # extract the borders
+            p = pdf[borders]
+        elif len(pdf) == len(borders):
+            p = pdf
+        else:
+            raise ValueError("Invalid size for probability.")
+        sampled_in_border = _get_samples(p)
+
+    else:
+        raise ValueError("Unsupported value for pdf use any of . ")
+        # TODO: allow custom pdf as argument (vector or function.)
+
+    line_locs = np.array(sorted(center_indexes + sampled_in_border))
+    # apply order of lines
+    if order == VDSorder.CENTER_OUT:
+        line_locs = _flip2center(sorted(line_locs), dim_size // 2)
+    elif order == VDSorder.RANDOM:
+        line_locs = rng.permutation(line_locs)
+    elif order == VDSorder.TOP_DOWN:
+        line_locs = np.array(sorted(line_locs))
+    else:
+        raise ValueError(f"Unknown direction '{order}'.")
+    return (line_locs / dim_size) - 0.5  # rescale to [-0.5,0.5]
+
+
+def get_random_loc_nd(N_samples, shape, pdf):
+    """Get random location in N-dimensions."""
+    # TODO
+    pass
+
+
+def stack_random(
+    traj,
+    dim_size,
+    center_prop,
+    accel=4,
+    pdf="uniform",
+    rng=None,
+    order="center-out",
```

Comment:
As mentioned in another comment, it would be nice to (later) have ordering tools independently of those functions as an additional step. Similarly, some trajectories currently have an oversampling internally defined while it should just always be a step after the trajectory initialization through `tools.oversample` added in PR #206. WDYT ?

Reply:
having a single oversampling function can be nice indeed.